static_frame/core/index_hierarchy.py

```diff
@@ -1413,67 +2082,83 @@ def equals(self,
             {compare_class}
             {skipna}
         '''
-        # NOTE: do not need to udpate array cache, as can compare elemetns in levels
+        # NOTE: do not need to udpate array cache, as can compare elements in levels
         if id(other) == id(self):
             return True
 
         if compare_class and self.__class__ != other.__class__:
             return False
-        elif not isinstance(other, IndexHierarchy):
+
+        if not isinstance(other, IndexHierarchy):
             return False
 
         # same type from here
         if self.shape != other.shape:
             return False
+
         if compare_name and self.name != other.name:
             return False
 
-        return self._levels.equals(other._levels, #type: ignore
-                compare_name=compare_name,
-                compare_dtype=compare_dtype,
-                compare_class=compare_class,
-                skipna=skipna,
-                )
+        if compare_dtype and not self.dtypes.equals(other.dtypes):
+            return False
+
+        if compare_class:
+            for self_index, other_index in zip(self._indices, other._indices):
+                if self_index.__class__ != other_index.__class__:
+                    return False
+
+        # indices & indexers are encoded in values_at_depth
+        for i in range(self.depth):
+            if not arrays_equal(
+                    self.values_at_depth(i),
+                    other.values_at_depth(i),
+                    skipna=skipna,
+                    ):
+                return False
+
+        return True
 
     @doc_inject(selector='sort')
     def sort(self: IH,
             *,
             ascending: BoolOrBools = True,
             kind: str = DEFAULT_SORT_KIND,
-            key: tp.Optional[tp.Callable[['IndexHierarchy'], tp.Union[np.ndarray, 'IndexHierarchy']]] = None,
+            key: tp.Optional[tp.Callable[[IH], tp.Union[np.ndarray, IH]]] = None,
             ) -> IH:
-        '''Return a new Index with the labels sorted.
+        '''
+        Return a new Index with the labels sorted.
 
         Args:
             {ascendings}
             {kind}
             {key}
         '''
         if self._recache:
-            self._update_array_cache()
+            self._update_array_cache() # pragma: no cover
```

Comment:
Why skip coverage of this scenario?

Reply:
See this response: https://github.com/InvestmentSystems/static-frame/pull/437#discussion_r831501537